example/pug-lang/src/main.c

```diff
@@ -0,0 +1,194 @@
+/* -*- coding: utf-8-unix -*- */
+
+#define CPARSEC_CONFIG_IMPLEMENT /* generate parsec code if defined */
+
+#include "interpreter/interpreter.h"
+#include "parser/stmt.h"
+
+struct Options {
+  bool use_color;
+  bool self_test;
+  CharBuff script;
+} options;
+
+struct Options pug_getopts(int argc, char** argv);
+void pug_help(String argv0);
+bool pug_parseTest(String input);
+void pug_self_test(void);
+
+// -----------------------------------------------------------------------
+int main(int argc, char** argv) {
+  options = pug_getopts(argc, argv);
+  if (options.self_test) {
+    pug_self_test();
+  }
+  if (options.script.length) {
+    return !pug_parseTest(options.script.data);
+  }
+
+  return 0;
+}
+
+// -----------------------------------------------------------------------
+#ifdef _WIN32
+/* for Windows platform (including Mingw) */
+static bool is_ANSI_escape_available(void) {
+  return false;
+}
+#else
+/* for Other platform (including MSYS) */
+#include <unistd.h>
+static bool is_ANSI_escape_available(void) {
+  return isatty(fileno(stdout));
+}
+#endif
+
+#define BOLD ";1"
+#define DIM ";2"
+#define UNDERLINE ";4"
+#define BLACK ";30"
+#define RED ";31"
+#define GREEN ";32"
+#define YELLOW ";33"
+#define BLUE ";34"
+#define MAGENTA ";35"
+#define CYAN ";36"
+#define WHITE ";37"
+
+static void eprintf(const char* restrict esc, const char* restrict fmt,
+                    ...) {
+  va_list ap;
+  va_start(ap, fmt);
+  if (!options.use_color) {
+    vprintf(fmt, ap);
+  } else {
+    printf("\033[0%sm", esc);
+    vprintf(fmt, ap);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [If format strings can be influenced by an attacker, they can be exploited (CWE-134).  Use a constant for the format specification. ](https://app.codacy.com/manual/mori0091/cparsec3/pullRequest?prid=5329824)

Reply:
All format strings passed to the function is a constant (string literal), for now.
In other words, there is no case that a user-input or a non-constant string is passed to `vprintf()` as a format string.